doc/Code/Library.org

```diff
@@ -1467,6 +1467,113 @@ Can be added to via core translation
   + [[Traverse]]
   + [[Library]]
   + [[Juvix/Pipeline]]
+* Sexp
+** Setup <<Sexp/Setup>>
+** src
+*** Juvix
+**** Sexp <<Juvix/Sexp>>
+This module serves as the main sexpression import it contains the
+sexp type and all the various helper functionality one can need
+- _Relies on_
+  + [[Library]]
+  + [[Library]]
+  + [[NameSymbol]]
+  + [[Juvix/Sexp/Parser]]
+  + [[Sexp/Types]]
+***** Parser <<Juvix/Sexp/Parser>>
+- _Relies on_
+  + [[Library]]
+  + [[NameSymbol]]
+  + [[Library/Parser]]
+  + [[Parser/Internal]]
+  + [[Sexp/Types]]
+***** Types <<Sexp/Types>>
+- _Relies on_
+  + [[Library]]
+  + [[LineNum]]
+  + [[NameSymbol]]
+***** Structure
+****** Frontend <<Structure/Frontend>>
+Hardeing S-expressions into a more readable form. Here we use a
+mixture of record structures and aliases. Each cover a form that we
+wish to talk about rather than just match away at
+- _The form for transformation follows this structure_
+#+begin_src haskell
+  -- the data type
+  data Form = ... deriving (Show)
+  is<Form>   :: Sexp.T -> Bool
+  to<Form>   :: Sexp.T -> Maybe <Form>
+  from<Form> :: <Form> -> Sexp.T
+#+end_src
+_TODO_
+ 1. Figure out if we can even express a spec system in
+    Haskell... =to<Form>= and =From<From>= have the exact same signature
+ 2. replace the repeat code with the =to<Form>= with an abstraction
+ 3. put the meta data with the form so we don't have to do it by
+    hand in the code that uses this
+    1. Use =Juvix.Library.LineNum=
+    2. append the =Form= With this
+    3. have to<Form> fill this
+    4. Have extra smart consturctors that are =<form>=, so that we
+       can automatically fill in this meta data
+- _Relies on_
+  + [[Library]]
+  + [[NameSymbol]]
+  + [[Juvix/Sexp]]
+  + [[Structure/Helpers]]
+****** Helpers <<Structure/Helpers>>
+- _Relies on_
+  + [[Library]]
+  + [[Juvix/Sexp]]
+****** Lens
+- _Relies on_
+  + [[Structure/Frontend]]
+  + [[Transition]]
+****** Transition
+Hardeing S-expressions into a more readable form. Here we use a
+mixture of record structures and aliases. Each cover a form that we
+wish to talk about rather than just match away at
+- _The form for transformation follows this structure_
+#+begin_src haskell
+  -- the data type
+  data Form = ... deriving (Show)
+  is<Form>   :: Sexp.T -> Bool
+  to<Form>   :: Sexp.T -> Maybe <Form>
+  from<Form> :: <Form> -> Sexp.T
+#+end_src
+_TODO_
+ 1. Figure out if we can even express a spec system in
+    Haskell... =to<Form>= and =From<From>= have the exact same signature
+ 2. replace the repeat code with the =to<Form>= with an abstraction
+ 3. put the meta data with the form so we don't have to do it by
+    hand in the code that uses this
+    1. Use =Juvix.Library.LineNum=
+    2. append the =Form= With this
+    3. have to<Form> fill this
+    4. Have extra smart consturctors that are =<form>=, so that we
+       can automatically fill in this meta data
```

Comment:
This comment looks like it's repeated between `Frontend` and `Transition` -- should it be in a single place?

Reply:
perhaps but for now it should stay like this I think?